firmware/spade/src/rpi/main.c

```diff
@@ -265,74 +285,173 @@ int main() {
+    multicore_reset_core1();
 
-    load_new_scripts();
-  }
+    update_save_version(); // init here to avoid irqs on other core
 
-  // Start a core to listen for keypresses.
-  multicore_launch_core1(core1_entry);
+    multicore_launch_core1(core1_entry);
 
-  /**
-   * We get a bunch of fake keypresses at startup, so we need to
-   * drain them from the FIFO queue.
-   *
-   * What really needs to be done here is to have button_init
-   * record when it starts so that we can ignore keypresses after
-   * that timestamp.
-   */
-  sleep_ms(50);
-  while (multicore_fifo_rvalid()) multicore_fifo_pop_blocking();
+    /**
+       * We get a bunch of fake keypresses at startup, so we need to
+       * drain them from the FIFO queue.
+       *
+       * What really needs to be done here is to have button_init
+       * record when it starts so that we can ignore keypresses after
+       * that timestamp.
+       */
+    sleep_ms(50);
+    while (multicore_fifo_rvalid()) multicore_fifo_pop_blocking();
 
-  /**
-   * Wait for a keypress to start the game.
-   *
-   * This is important so games with e.g. infinite loops don't
-   * brick the device as soon as they start up.
-   */
-  while(!multicore_fifo_rvalid()) {
-    strcpy(errorbuf, "                    \n"
-                     "                    \n"
-                     "                    \n"
-                     "                    \n"
-                     "                    \n"
-                     "                    \n"
-                     "                    \n"
-                     "  PRESS ANY BUTTON  \n"
-                     "       TO RUN       \n"
-                     "                    \n"
-                     "                    \n"
-                     "                    \n"
-                     "                    \n"
-                     "                    \n"
-                     " sprig.hackclub.com \n");
-    render_errorbuf();
-    st7735_fill_start();
-    render(st7735_fill_send);
-    st7735_fill_finish();
+    int games_i = 0;
 
-    load_new_scripts();
-  }
+    int games_len = 8;
+    Game* games = malloc(games_len * sizeof(Game));
+
+    Welcome_Screen welcome_state = NEW_SLOT;
+
+    for (;;) {
+
+        if (games_i >= games_len && games_i != 0) {
+            games_i = games_len - 1;
+        }
+
+        if (welcome_state == DELETE_CONFIRM) {
+            // no-op
+        } else if (games_len == 0) {
+            welcome_state = NEW_SLOT;
+        } else {
+            welcome_state = RUN_GAME;
+            set_game(games[games_i]);
+        }
+
+        /**
+        * Wait for a keypress to start the game.
+        * This is important so games with e.g. infinite loops don't
+        * brick the device as soon as they start up.
+        */
+
+        if (multicore_fifo_rvalid()) {
+            uint32_t c = multicore_fifo_pop_blocking();
+
+            if (welcome_state == DELETE_CONFIRM) {
```

Comment:
suggest moving logic for determining welcome state into separate method

Reply:
can't really do this since it relies on the games list to change index, etc.